dotnet/samples/Demos/ModelContextProtocolClientServer/MCPServer/Tools/MailboxUtils.cs

```diff
@@ -0,0 +1,129 @@
+            Attachments = [EmbeddedResource.ReadAsBytes("SalesReport2014.png")]
+        };
+
+        var email2 = new Email
+        {
+            Sender = "hr.department@example.com",
+            Subject = "Employee Birthdays and Positions",
+            Body = "Attached is the list of employee birthdays and their positions. Please check it and let me know of any updates by tomorrow." +
+                   "Also, we're planning a hike this Sunday morning. It would be great if you could join us. Let me know if you're interested!",
+            Attachments = [EmbeddedResource.ReadAsBytes("EmployeeBirthdaysAndPositions.png")]
+        };
+
+        CreateMessageRequestParams request = new()
+        {
+            SystemPrompt = "You are a helpful assistant. You will be provided with a list of emails. Please summarize them. Each email is followed by its attachments.",
+            Messages = CreateMessagesFromEmails(email1, email2),
+            Temperature = 0
+        };
+
+        // Send the sampling request to the client to summarize the emails
+        CreateMessageResult result = await server.RequestSamplingAsync(request, cancellationToken: CancellationToken.None);
+
+        // Assuming the response is a text message
+        return result.Content.Text!;
+    }
+
+    /// <summary>
+    /// Creates a list of SamplingMessage objects from a list of emails.
+    /// </summary>
+    /// <param name="emails">The list of emails.</param>
+    /// <returns>A list of SamplingMessage objects.</returns>
+    private static List<SamplingMessage> CreateMessagesFromEmails(params Email[] emails)
+    {
+        var messages = new List<SamplingMessage>();
+
+        foreach (var email in emails)
+        {
+            messages.Add(new SamplingMessage
+            {
+                Role = Role.User,
+                Content = new Content
+                {
+                    Text = $"Email from {email.Sender} with subject {email.Subject}. Body: {email.Body}",
+                    Type = "text",
+                    MimeType = "text/plain"
+                }
+            });
+
+            if (email.Attachments != null && email.Attachments.Count != 0)
+            {
+                foreach (var attachment in email.Attachments)
+                {
+                    messages.Add(new SamplingMessage
+                    {
+                        Role = Role.User,
+                        Content = new Content
+                        {
+                            Type = "image",
+                            Data = Convert.ToBase64String(attachment),
+                            MimeType = "image/png",
+                        }
+                    });
+                }
+            }
+        }
+
+        return messages;
+    }
+}
+
+/// <summary>
+/// Represents an email.
+/// </summary>
+public class Email
```

Comment:
Inner class of the tool, or different file within `Models`.
```suggestion
public class Email
```